src/python/zquantum/core/measurement.py

```diff
@@ -500,7 +514,42 @@ def get_expectation_values(
                     value = -float(count) / num_measurements
                 expectation += np.real(coefficient) * value
             expectation_values.append(np.real(expectation))
-        return ExpectationValues(np.array(expectation_values))
+
+        correlations = np.zeros((len(ising_operator.terms),) * 2)
+        for i, first_term in enumerate(ising_operator.terms):
+            for j, second_term in enumerate(ising_operator.terms):
+                first_term_qubits = set(op[0] for op in first_term)
+                second_term_qubits = set(op[0] for op in second_term)
+                marked_qubits = first_term_qubits.symmetric_difference(
+                    second_term_qubits
+                )
+                for bitstring, count in bitstring_frequencies.items():
+                    bitstring_int = convert_bitstring_to_int(bitstring)
+                    if parity_even_p(bitstring_int, marked_qubits):
+                        value = float(count) / num_measurements
+                    else:
+                        value = -float(count) / num_measurements
+                    correlations[i, j] += (
+                        np.real(
+                            ising_operator.terms[first_term]
+                            * ising_operator.terms[second_term]
+                        )
+                        * value
+                    )
+
+        denominator = (
+            num_measurements - 1 if use_bessel_correction else num_measurements
+        )
+        covariances = np.zeros((len(ising_operator.terms),) * 2)
```

Comment:
`covariances = correlations - expectation_values[:, np.newaxis] *  expectation_values[np.newaxis, :]` should be faster and more compact.

Reply:
Done!

Reply:
Thanks!